feat(surveys): `posthog` changes to support valid HTML question rendering

dmarticus opened the pull request:
## Summary

I modified the survey renderer to differentiate between `Text` and `HTML` references when writing question descriptions.  Before this change, we had a bug where users could bypass the HTML question description markup paywall by writing raw HTML into their Text description (and it would still be interpolated as HTML if the tags were valid), so this change makes it so that we only interpolate HTML in the popup survey if the HTML feature is (a) paid for and (b) enabled.

### Important Note

This PR includes changes to the `Survey` Model, which is a Core model that is serialized throughout our app and stored in our Postgres tables.  Specifically, I modified the `BasicSurveyQuestion` and `SurveyAppearance` models to store the content type of the `description fields.  

Schemas before
```
export interface SurveyQuestionBase {
    question: string
    description?: string | null
    optional?: boolean
    buttonText?: string
}

export interface SurveyAppearance {
    backgroundColor?: string
    submitButtonColor?: string
    // TODO: remove submitButtonText in favor of buttonText once it's more deprecated
    submitButtonText?: string
    ratingButtonColor?: string
    ratingButtonActiveColor?: string
    borderColor?: string
    placeholder?: string
    whiteLabel?: boolean
    displayThankYouMessage?: boolean
    thankYouMessageHeader?: string
    thankYouMessageDescription?: string
    autoDisappear?: boolean
    position?: string
    // widget only
    widgetType?: 'button' | 'tab' | 'selector'
    widgetSelector?: string
    widgetLabel?: string
    widgetColor?: string
    shuffleQuestions?: boolean
}
```

Schemas after
```
export interface SurveyQuestionBase {
    question: string
    description?: string | null
    descriptionContentType?: SurveyQuestionDescriptionContentType // new field
    optional?: boolean
    buttonText?: string
}

export interface SurveyAppearance {
    backgroundColor?: string
    submitButtonColor?: string
    // TODO: remove submitButtonText in favor of buttonText once it's more deprecated
    submitButtonText?: string
    ratingButtonColor?: string
    ratingButtonActiveColor?: string
    borderColor?: string
    placeholder?: string
    whiteLabel?: boolean
    displayThankYouMessage?: boolean
    thankYouMessageHeader?: string
    thankYouMessageDescription?: string
    thankYouMessageDescriptionContentType?: SurveyQuestionDescriptionContentType // new field
    autoDisappear?: boolean
    position?: string
    // widget only
    widgetType?: 'button' | 'tab' | 'selector'
    widgetSelector?: string
    widgetLabel?: string
    widgetColor?: string
    shuffleQuestions?: boolean
}
```

Given that we now have new fields that posthog-js requires to differentiate between how to render descriptions, we now need some strategy to backfill this data so that old surveys can render descriptions appropriately.  Fortunately, only one (1) user has the question description field even populated (and they're using plain text), and no one uses this field for the confirmation message description, so this is pretty safe.  Furthermore, in my `posthog-js` changes, [I elected to default to `Text` rendering](https://github.com/PostHog/posthog-js/pull/1206/files#diff-ca75b1ebdb1e4a75293f5a5f4d0bdc7b94c5ca7243b5ac9ebb49147a6dd34d83R229-R231).

In summary, this feature isn't really used very much (yet), which is good because this makes a change like this pretty safe.

### More info

Needs to ship with the associated `posthog-js` changes here: https://github.com/PostHog/posthog-js/pull/1206/

Closes: https://github.com/PostHog/posthog/issues/22509

## Changes

See the ticket for how the HTML parsing behavior used to look.

New behavior looks like this: 

![2024-05-26 13 25 09](https://github.com/PostHog/posthog/assets/4853149/e583f685-b11e-46b3-99da-e84855d5acaa)

## Does this work well for both Cloud and self-hosted?

Yes

## How did you test this code?

### Manual Test Cases

(Ran these tests for every question field and for the thank you message description)
- Text renders correctly
- HTML renders correctly
- Switching between text and HTML causes the tags to be manually printed (if text) and interpolated (if HTML)
- 
<!-- Include automated tests if possible, otherwise describe the manual testing routine. -->

dmarticus committed:
adds a beforeUnload condition to the survey editing experience so that users don't accidentally refresh the page and lose their survey edit progress.  Also fixes an issue where the initial survey creation process never set the isEditingSurvey state to true.

dmarticus committed:
remove this footer, it's superfluous if we have a sticker header

dmarticus committed:
omg linter chill I'm DEBUGGING

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
Merge branch 'master' into feat/add-beforeunload-warning-to-survey-page

dmarticus committed:
handle the  button logic, and fix the issue where the editing sessions were incorrectly persisting

dmarticus committed:
Merge branch 'feat/add-beforeunload-warning-to-survey-page' of github.com:PostHog/posthog into feat/add-beforeunload-warning-to-survey-page

dmarticus committed:
remove old log

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

dmarticus committed:
remove console logs

dmarticus committed:
add copy for survey wait period

dmarticus committed:
Merge branch 'fix/survey-release-summary-needs-default-text' into feat/survey-popup-ui-polish

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

dmarticus committed:
don't use local posthog-js

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

dmarticus committed:
use published version of posthog-js

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus committed:
ah it was the LOCKfile

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

neilkakkar commented:
Heya! How did you figure out there's only one user using html descriptions? This doesn't sound right, I'm pretty sure there's multiple just in our team, and I'm _very_ sure I've seen multiple customers with html surveys during support.

One other thing worth noting is that this same problem applies not just to descriptions, but the question text, and button text as well - they're all rendered as HTML always.

I love what you're doing here, I think we should extend this to a global survey render type setting: If enabled, and set, we render all fields as HTML. If not, we render all fields as text. I.e. it's not just for description.

---

For backwards compatibility, and not breaking existing running surveys, I'd recommend defaulting to HTML for all surveys where this is not set. Plus, for all new surveys, we always make sure to set it.

Oh, and also, we'd want API level validation here - if `html` isn't allowed, anyone using the API to create a new survey also shouldn't be allowed to set `html` as the render type

dmarticus committed:
don't include this beforeunload stuff that is buggy

dmarticus committed:
clean more stuff

dmarticus committed:
resolve merge conflicts?

dmarticus commented:
@neilkakkar, you're absolutely right, there was a bug in my postgres syntax.  I amended it today and found that there were [101 surveys that had either HTML](http://metabase-prod-us/question/681-surveys-using-html-descriptions-in-either-the-questions-or-confirmation-messages) in the question description or confirmation message description (in case you can't load my metabase dashboard, here's the query)
```sql
SELECT 
    COUNT(*) 
FROM (
    SELECT 
        id
    FROM 
        posthog_survey,
        jsonb_array_elements(questions) AS elem
    WHERE 
        elem->>'description' IS NOT NULL
        AND elem->>'description' <> ''
        AND elem->>'description' ~ '<[^>]+>'
    UNION
    SELECT 
        id
    FROM 
        posthog_survey
    WHERE 
        appearance ? 'thankYouMessageDescription'
        AND appearance->>'thankYouMessageDescription' IS NOT NULL
        AND appearance->>'thankYouMessageDescription' <> ''
        AND appearance->>'thankYouMessageDescription' ~ '<[^>]+>'
) AS combined_results;
```
and [49 in EU](http://metabase-eu/question/157-surveys-using-html-descriptions-in-either-the-questions-or-confirmation-messages), for a total of 150 surveys.  So we'll definitely need to figure out a backfill strategy that doesn't bork all of these existing surveys that are marked up.

To your point on not wanting to break existing surveys, I'm fine defaulting to HTML if the value was never set in the survey itself.  I'll make that change in `posthog-js` (since that's where the rendering actually happens), and then I'll make that new and updated surveys start to save the content type in the survey definition.

dmarticus committed:
default to text for the editor experience of new surveys if no values exist

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

dmarticus committed:
commit this for now

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus committed:
resolving merge conflicts I hope

dmarticus committed:
remove log to satisfy the build

dmarticus committed:
oh jeez this isn't good

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
drawing the rest of the owl

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus committed:
need to implement these changes too

dmarticus committed:
use my type everywhere

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
some bad merge resolutions

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus commented:
@neilkakkar I made more progress on this today but there's still a few things I'd to discuss with you before I go much further.

- I'm using `useEffect()` in the two components parent components where I use the `HTMLEditor` component ([example](https://github.com/PostHog/posthog/pull/22533/files#diff-a12eb517d1b5fef23ebe3b6933695b3c304f8a4c14c436becb1cbc26604acad5R91-R97)), which I know we shouldn't do. I tried using kea logic to do this instead (here the [action](https://github.com/PostHog/posthog/pull/22533/files#diff-0a051008afcf371e7c0f56a8f251f7d5fd2b20c2c8b491d1b4bd157347e593c9R149-R152) and [reducer](https://github.com/PostHog/posthog/pull/22533/files#diff-0a051008afcf371e7c0f56a8f251f7d5fd2b20c2c8b491d1b4bd157347e593c9R513-R525) what I added in my attempt to do this), but I wasn't able to get changes to persist across clicks; any time I clicked a different tab, it just reverted back to the default.  I'd love to understand what my approach should be here to handle managing state where it involves changing the values of the survey on the page, too.
- I'm not sure exactly where the HTML validation on the backend is happening.  I'm trying to answer this question:

> Oh, and also, we'd want API level validation here - if html isn't allowed, anyone using the API to create a new survey also shouldn't be allowed to set html as the render type

but I don't know where to look for this just yet.  

- Finally, I'm not really sure what sorts of tests I should be writing here, but I feel like this is a significant enough change that I'd like to write them.  Would love to get your opinion on how to approach this.

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
new changes

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus committed:
fix formatter

dmarticus committed:
more ruff formatting

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

dmarticus committed:
adding some stories, wahey

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
added tests for the new API behavior

dmarticus committed:
mErge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
Merge branch 'master' into feat/survey-popup-ui-polish

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus commented on posthog/api/test/test_survey.py:
```diff
@@ -1,28 +1,28 @@
-import re
-from rest_framework import status
 from django.core.cache import cache
 from django.test.client import Client
 from freezegun.api import freeze_time
+from rest_framework import status
+
 from posthog.api.survey import nh3_clean_with_allow_list
+from posthog.constants import AvailableFeature
+from posthog.models import FeatureFlag
 from posthog.models.cohort.cohort import Cohort
-
 from posthog.models.feedback.survey import Survey
 from posthog.test.base import (
     APIBaseTest,
-    ClickhouseTestMixin,
     BaseTest,
+    ClickhouseTestMixin,
     QueryMatchingTest,
-    snapshot_postgres_queries,
-    snapshot_clickhouse_queries,
     _create_event,
+    snapshot_clickhouse_queries,
+    snapshot_postgres_queries,
 )
 
-from posthog.models import FeatureFlag
-
```
ooof, that was an auto-reordering from my editor.  Should I undo this?

dmarticus committed:
bring back og test

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

dmarticus committed:
a line

dmarticus commented on posthog/test/base.py:
```diff
@@ -351,14 +348,28 @@ def setUp(self):
-        # Sets the cloud mode to stabilise things tests, especially num query counts
-        # Clear the is_rate_limit lru_Caches so that they does not flap in test snapshots
+        # Sets the cloud mode to stabilize things tests, especially num query counts
+        # Clear the is_rate_limit lru_Caches so that they do not flap in test snapshots
```
flyby grammar policing 🚓

neilkakkar replied:
Americans 💪 😅  (but yes thank you our default is American English no matter how much British English I try to sneak in 😂 )

dmarticus commented on posthog/test/base.py:
```diff
@@ -351,14 +348,28 @@ def setUp(self):
         if self.CONFIG_AUTO_LOGIN and self.user:
             self.client.force_login(self.user)
 
+    def create_organization_with_features(self, features):
+        organization = Organization.objects.create(name="Test Organization")
+        organization.available_product_features = [{"name": feature, "key": feature} for feature in features]
+        organization.save()
+        return organization
+
+    def create_team_with_organization(self, organization):
+        return Team.objects.create(organization=organization, name="Test Team")
+
+    def create_user_with_organization(self, organization):
+        user = User.objects.create_user(email="testuser@example.com", first_name="Test", password="password")
+        organization.members.add(user)
+        return user
+
```
added some functionality to our APIBaseTest class to support adding/removing paid features.

neilkakkar replied:
very nice

dmarticus committed:
self-review feedback

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
messed up formatting

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

dmarticus committed:
rewrote the components to handle differentiating between tabs and survey content

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

github-actions[bot] committed:
Update UI snapshots for `chromium` (1)

dmarticus committed:
a few more small tings

dmarticus committed:
Merge branch 'feat/survey-popup-ui-polish' of github.com:PostHog/posthog into feat/survey-popup-ui-polish

neilkakkar commented on frontend/src/scenes/surveys/SurveyAppearanceUtils.tsx:
```diff
@@ -50,22 +51,36 @@ export function PresentationTypeCard({
     textPlaceholder?: string
 }): JSX.Element {
     const { surveysHTMLAvailable } = useValues(surveysLogic)
+    const { activeTabKey } = useValues(surveyLogic)
```
Ah yeah this is becoming problematic - it doesn't set the actual value in the survey, but sets it in the UI, making them de-sync 😬

neilkakkar replied:
I pushed a proposal, let me know what you think! DIdn't test this with the permissions for html though - so not sure if the flow there works well.

neilkakkar committed:
proposal for content type setting

neilkakkar reviewed (commented):
One comment, keen to hear your thoughts

dmarticus committed:
Feat/html rendering (#22698)

* feat: move query performance polling to its own celery task in a performant manner (#22497)

* feat(hogql): inline filters into subqueries (#22468)

* feat(hogql): ignore future persons (#22507)

* fix: add persistence to redis and kafka on hobby (#22563)

* add persistence to redis and kafka on hobby

This is to address https://github.com/PostHog/posthog/issues/22291, but potentially fixes other redis/kafka related conditions over containers updates / rebuilds

* proper path to kafka data volume

* add volume definitions

---------

Co-authored-by: Frank Hamand <frank@posthog.com>

* chore(vscode): adapt celery configuration (#22606)

* fix: Missing scope on events endpoint (#22604)

* revert: Make port a tel (#22609)

* chore: add ids to admin for qol when debugging/doing support (#22607)

add ids to admin for qol when debugging/doing support

* feat: add person profiles instructions to SDKs (#22403)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* fix(insights): fix stacked line chart (#22598)

* fix: filter width (#22615)

* feat(data-warehouse): data imports pipeline UI (#22553)

* add data warehouse source tables into pipeline ui

* add empty state

* types

* ui

* total rows synced column

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* add tooltip

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* remove unnecessary deleteRecordSuccess call in managed proxy (#22475)

this was just to restart polling back in the dark ages before we always
ran the poll check

* fix(data-warehouse): Dont pass password if its not set (#22619)

Dont pass password if its not set

* chore(deps): Update posthog-js to 1.136.4 (#22630)

* fix(data-warehouse): Fix getting ssh enabled var (#22629)

Fix getting ssh enabled var

* chore: add a data attr (#22621)

* feat(hogql): type system (#22587)

* restore wip

* add numeric operator signatures

* Fix all the unit tests

* Improvements to type system

* Updatd mypy

* Fixed mypy issues

* Fixed property types

* Fixed mypy issues

---------

Co-authored-by: eric <eeoneric@gmail.com>

* feat(data-warehouse): Resolve expression fields underlying type in db schema query (#22611)

* restore wip

* add numeric operator signatures

* Fix all the unit tests

* Improvements to type system

* Updatd mypy

* Fixed mypy issues

* Fixed property types

* Fixed mypy issues

* Resolve expression fields to their underlying types during the database schema query

---------

Co-authored-by: eric <eeoneric@gmail.com>

* fix(web-analytics): Fix searching for session properties with multiple words in the search term (#22632)

* chore: Add `query-async` to `PERSISTED_FEATURE_FLAGS` (#22628)

* chore(insights): Consolidate HogQL flags to `hogql-insights-preview` (#22631)

* fix(flags): Better code snippets (#22589)

* fix(hogql): derive cache keys from pydantic model_dump (#22465)

* feat: Timeout row count if it takes longer than 1/12 of frequency (#22636)

* feat: Timeout row count if it takes longer than 1/12 of frequency

* fix: Don't fail silently

* chore: add two tests (#22641)

* feat: Remove FF code for enabling sr in pa onboarding (#22643)

* remove ff code

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* chore(deps): Update posthog-js to 1.136.5 (#22642)

Co-authored-by: posthog-bot <posthog-bot@users.noreply.github.com>

* chore: remove addon confirm modal and show message instead (#22616)

* Remove addon confirm modal and show message instead

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update BillingProductAddon.tsx

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update billing mock date

* Update BillingProductAddon.tsx

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* fix: add a product check first for some features (#22581)

* add a product check first for some features

* Update payGateMiniLogic.tsx

* Update payGateMiniLogic.tsx

* Update payGateMiniLogic.tsx

* fix: Patch the usage_summary value for rows_synced (#22600)

* patch the usage_summary value for rows_synced

* break up code for readability

* chore: remove the email-verification-ticket-submission ff (#22639)

* Remove the email-verification-ticket-submission ff

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* chore: remove onboarding reverse proxy ff (#22638)

* turn off the onboarding reverse proxy ff

* Update onboardingLogic.tsx

* Update onboardingLogic.tsx

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* fix(billing UI): fix success color (#22634)

* perf: Rewrite retention query to use arrays instead of joins (#22521)

* perf: Rewrite retention query to use arrays instead of joins

* Fix all events

* fix types and tests

* fix baseline

* fix first time retention weeks

* Update query snapshots

* fix basline

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* fix: enable compression for reverse_proxy responses (#22237)

enable compression for reverse_proxy responses

compression is disabled in caddy by default, rendering frontend PAINFULLY slow (as if Django is not slow enough already)

* fix: limit kafka log retention to 1h by default (#22270)

limit kafka log retention to 1h by default

Co-authored-by: Frank Hamand <frankhamand@gmail.com>

* feat: live events feed (#22302)

* initial commit

* initial commit

* fix up some types

* Add team id

* add client side filters

* check live events in onboarding

* add eventsource

* clean up live table logic

* add event source module

* Delete eventsManagementDescribers.tsx

* update event source usage

* Update liveEventsTableLogic.ts

* Update UI snapshots for `chromium` (2)

* add team live events token

* Delete liveEventsTableLogic.ts

* Update types.ts

* switch to use window event source

* improvements / feature flags

* cleanup

* update the live event host

* Update UI snapshots for `chromium` (2)

* remove event source lib

* fix up event source types

* Clean up live events view

* Delete eventsManagement.ts

* Update SDKs.tsx

* improve live event typing

* add better loading for the table

* update the live events table columns

* add last batch timestamp check

* add toast for error

* rename events management to activity

* Hookup proper team id

* Update start

* Fix types

* Update some tests

* Put SDKs back with no live event changes

* Update verifiedDomainsLogic.test.ts.snap

* Update verifiedDomainsLogic.test.ts.snap

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Update query snapshots

* Update UI snapshots for `chromium` (2)

* Use `preserveParams()` in redirect from old URL

* Clean up UI and refactor tabs

* Update E2E tests

* Update UI snapshots for `chromium` (2)

* Don't hide "Reload" when live events available

* Remove unused import

* Update UI snapshots for `chromium` (2)

* Improve local batching reliability

* Make console error clearer

* Clarify directory structure

* Update UI snapshots for `chromium` (2)

* Jot down source of `EventSource` type

* Remove unused scene code

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update a11.cy.ts

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* Remove any effects for users with flag off

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Michael Matloka <michal@matloka.com>
Co-authored-by: Michael Matloka <dev@twixes.com>

* chore(insights): remove flag hogql-insights-preview (#22651)

* remove flag

* cleanup

* killing code

* update image exporter

* mypy baseline

* test_insight_cache

* kill tests that don't work anymore

* remove settings

* test passes

* test pass

* mypy fixes

* remove commented code

* rename and add capture back in

* remove references to feature flag

* remove the falg

* comment out failing test

* Update UI snapshots for `chromium` (1)

* fix jest

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (1)

* Fix storybook test

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: Alexander Spicer <aspicer@gmail.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Tom Owers <owerstom@gmail.com>

* feat(hog): dicts and arrays (#22618)

* fix: if a hedegehog falls in the woods (#22658)

* feat: add ability to sort path filter rules (#22633)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* fix(exports): Fix export to use process_query_dict (#22664)

* feat: Pipeline UI: metrics page use insights like date selection (#22601)

* fix: console warning from the toolbar (#22660)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* feat: Improve materialize command (#22666)

* chore(data-warehouse): update states (#22580)

* add data warehouse source tables into pipeline ui

* add empty state

* types

* ui

* total rows synced column

* use active instead of completed and failed is error

* Update UI snapshots for `chromium` (2)

* Update UI snapshots for `chromium` (2)

* fix test

* change enum

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* feat: use new deployment trigger for temporal worker deployments (#22668)

use new deployment trigger for temporal worker deployments

these trigger a new workflow in posthog/charts which creates a statefile
commit instead of deploying with manually set values from env vars.

the statefile commit then triggers a deploy - this means 100% of our
deployment state is codified, simplifying rollbacks and deploys

* fix(hog): property assigment via dots (#22659)

* chore(surveys): refactor survey preview (#22617)

* chore: update error message in temporal worker (#22672)

this is just to trigger a temporal worker deploy to test the new
deployment process

* fix: Revert "feat: use new deployment trigger for temporal worker deployments" (#22677)

Revert "feat: use new deployment trigger for temporal worker deployments (#22…"

This reverts commit 0834410018bc2b7299d8169f828144c5e94b9f9e.

* chore: Move date fills around to use arrays (#22603)

* perf: Single query for funnels with breakdown

* also fix trends

* Update query snapshots

* Update query snapshots

* Update query snapshots

* fix test

* Update query snapshots

* Update query snapshots

* Update query snapshots

* fix

* Update query snapshots

* Update query snapshots

* Update query snapshots

* snapshots

* chore: Move date fill in trends queries around

* fix smoothing

* fix formula breakdown

* fix

* fix tests

* fix too many columns

* fix

* Update query snapshots

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* fix(properties): fix invalid property filters (#22656)

* chore(data-warehouse): add beta notice (#22646)

* add pricing notice

* Update UI snapshots for `chromium` (2)

* change pricing to beta notice

* Update UI snapshots for `chromium` (1)

* Update UI snapshots for `chromium` (2)

* change zendesk logo

* Update UI snapshots for `chromium` (2)

---------

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* chore(plugin-server): add distinct id batching metric (#22678)

* chore(deps): Update posthog-js to 1.136.7 (#22674)

* chore(data-warehouse): revert update states (#22681)

Revert "chore(data-warehouse): update states (#22580)"

This reverts commit 2dcef5cff9e79c962e2854170e74f4e40070ea89.

* fix(experiment): fix MDE modal insight query (#22680)

* feat: add a launch config for local billing (#22683)

* chore(vscode): add autoreload for celery (#22675)

* fix(insights): Fix dashboard export size again (#22671)

* feat(insights): make 'all' and multi cohort work in trends actors (#22624)

* feat: Current bill value should respect usage limits (#22597)

* have current usage dollar value respect billing limits

* add test

* update according to pr feedback

* feat: pipeline-ui-3000 batch export runs (#22552)

* fix: Add in checks on dictionary accessors when updating local usage information in posthog (#22688)

* fix the dictionary accessors

* lint

* fix: remove feature gate for action description (#22687)

Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>

* feat: change language on person profile popup (#22648)

* chore: prune unnecessary metrics from usage reports (#22693)

* fix(actions): display more than 100 actions (#22692)

* feat: allow viewing of activity side panel when impersonating (#22682)

* this all works for the non-locked mode, let's get it working for locked stuff now

* this handles all the necessary state management, but still renders the HTML

* this allows us to not render the HTML if the paywall up

---------

Co-authored-by: Sandy Spicer <sandy@posthog.com>
Co-authored-by: Marius Andra <marius.andra@gmail.com>
Co-authored-by: feedanal <111871756+feedanal@users.noreply.github.com>
Co-authored-by: Frank Hamand <frank@posthog.com>
Co-authored-by: Thomas Obermüller <thomas.obermueller@gmail.com>
Co-authored-by: Ben White <ben@posthog.com>
Co-authored-by: Tomás Farías Santana <tomas@tomasfarias.dev>
Co-authored-by: Zach Waterfield <zlwaterfield@gmail.com>
Co-authored-by: Raquel Smith <raquelmsmith@users.noreply.github.com>
Co-authored-by: github-actions <41898282+github-actions[bot]@users.noreply.github.com>
Co-authored-by: Eric Duong <eric@posthog.com>
Co-authored-by: Frank Hamand <frankhamand@gmail.com>
Co-authored-by: Tom Owers <tom@posthog.com>
Co-authored-by: PostHog Bot <69588470+posthog-bot@users.noreply.github.com>
Co-authored-by: Paul D'Ambra <paul@posthog.com>
Co-authored-by: eric <eeoneric@gmail.com>
Co-authored-by: Robbie <robbie.coomber@gmail.com>
Co-authored-by: Michael Matloka <dev@twixes.com>
Co-authored-by: Neil Kakkar <neilkakkar@gmail.com>
Co-authored-by: Bianca Yang <21014901+xrdt@users.noreply.github.com>
Co-authored-by: posthog-bot <posthog-bot@users.noreply.github.com>
Co-authored-by: Juraj Majerik <juro.majerik@gmail.com>
Co-authored-by: timgl <tim@glsr.nl>
Co-authored-by: Michael Matloka <michal@matloka.com>
Co-authored-by: Alexander Spicer <aspicer@gmail.com>
Co-authored-by: Tom Owers <owerstom@gmail.com>
Co-authored-by: David Newell <d.newell1@outlook.com>
Co-authored-by: Julian Bez <julian@posthog.com>
Co-authored-by: Tiina Turban <tiina303@gmail.com>
Co-authored-by: Brett Hoerner <brett@posthog.com>

dmarticus commented:
I accidentally made a bad merge into this branch, so I'm closing it in favor of https://github.com/PostHog/posthog/pull/22699
